### PR TITLE
Add simple token export behat test

### PR DESCRIPTION
**Pull request by pablothedude**
A simple token export test was added to have this covered.
https://www.pivotaltracker.com/n/projects/1163646/stories/171517126

### CODE REVIEW

File: tests/behat/features/ra_export.feature [hunk 1]
```diff
@@ -0,0 +1,28 @@
+Feature: A RAA manages tokens tokens registered in the selfservice portal
+  In order to manage tokens
+  As a RAA
+  I must be able to manage second factor tokens from my institution
```
Comment (MKodde) [inline tests/behat/features/ra_export.feature]: Please rewrite the feature outline to match the behavior that is actually tested in this feature.

File: tests/behat/features/ra_export.feature [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+  Scenario: RA user can't vet a token from another institution it is not RA for
```
Comment (MKodde) [inline tests/behat/features/ra_export.feature]: Please update scenario description to fit the scenario this feature covers.

File: tests/behat/features/ra_export.feature [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+      And the user "urn:collab:person:institution-a.example.com:jane-a-ra" has the role "ra" for institution "institution-a.example.com"
+      And the user "urn:collab:person:institution-a.example.com:jane-a-raa" has a vetted "yubikey"
+      And the user "urn:collab:person:institution-a.example.com:jane-a-raa" has the role "raa" for institution "institution-a.example.com"
+
+  Scenario: RA user can not export tokens
+    Given I am logged in into the ra portal as "jane-a-ra" with a "yubikey" token
+    When I visit the Tokens page
+    Then I should not see a token export button
+
+  Scenario: RA user can not export tokens
```
Comment (MKodde) [inline tests/behat/features/ra_export.feature]: Copy pasted without updating the scenario description?

**Comment by pablothedude** [comment]
I haven't added a test to cover the use case because I don't want the acceptance test to become to heavy. I also think that RA itself would be more suitable.
`If a user has RAA rights in at least one institution, the user is allowed to export all tokens for the institutions where the user has RA or RAA rights.`

**Comment by MKodde** [comment]
I don't see the logic in this fully. Having this kind of (often regression prone) features under test seems very useful to me. And doing so using a functional test seems a very logical thing to do. Yes, a unit test covering this would also be a possibility, but by the time we have that test coverage, we might be 2 or 3 releases down the road. My proposal is to either create a unit test in RA, or a scenario in the feature of this PR.

**Comment by pablothedude** [comment]
I agree that testing is useful but the Behat tests should be used to test interaction between the different Stepup projects. When and how an export button should be visible should be the responsibility of RA because that will be determined there and therefore should be tested there. That way the tests will also be easier to maintain in the long run..

Because there is no sufficient testing in RA and SA this is currently not done and seems the main reason why this bug was there in the first place. Currently the threshold seems too high to add a simple test there. I expect this would have been done initially.

Could you elaborate why a test wasn't added in the first place while it should have been tested according to the definition of done? And do you perhaps have some ideas how we could lower the threshold in the future so we could prevent this in the future from happening. I think there is much to gain here and I would like to have your suggestions (or fuel) on the pivotal issue dedicated to this.

https://www.pivotaltracker.com/story/show/171622570

For now I've added a basic test for the user story you mentioned, but we should address this TD as far as I'm concerned.

**Comment by MKodde** [comment]
Let's agree that we do not agree about the intent of the behat tests.

Thanks for adding the test.

No, I do not recall why I did not add tests to the exporter logic created in https://github.com/OpenConext/Stepup-RA/pull/143. The services at least could have been tested with little effort.

**Pull request merged**